apps/frontend/src/pages/projects/libs/components/project-card/project-card.tsx

```diff
@@ -31,11 +37,33 @@ const ProjectCard = ({
 		onDelete(project);
 	}, [onDelete, project]);
 
+	const currentDate = new Date();
+	const lastActivityDate = project.lastActivityDate
+		? new Date(project.lastActivityDate)
+		: null;
+
+	const daysDifference = lastActivityDate
+		? getDifferenceInDays(currentDate, lastActivityDate)
+		: null;
+
+	const lastUpdateLabel = lastActivityDate
+		? `Updated ${getRelativeDate(lastActivityDate, currentDate)}`
+		: null;
+
+	const colorStatus =
+		typeof daysDifference === "number"
+			? getActivityIndicatorStatus(daysDifference)
+			: null;
+
+	const hasActivityIndicator = lastUpdateLabel && colorStatus;
```

Comment:
hmm.. now we need to provide a fallback value when `lastUpdateLabel && colorStatus` are null. Is it okay?

Reply:
Or maybe we can write this like this:
```suggestion
	const hasActivityIndicator = 
		lastUpdateLabel !== null && 
		colorStatus !== null;
```

Reply:
`Boolean(lastUpdateLabel && colorStatus)` cannot return true for the null value. We need to convert it to boolean for sure and we don't want to write this condition inside the component render, because this was the whole point in moving it out. Either use the approach as @GvoFor suggested above, or cast the types, because typescript doesn't properly handle Boolean call, but we don't want to write additional checks because of typescript